packages/codegen/src/codegen.ts

```diff
@@ -0,0 +1,316 @@
+
+      const missingResponseSuffix = !name.toLowerCase().endsWith('response');
+      const schemaName = missingResponseSuffix ? name + 'Response' : name;
+      this.versionFile.components!.schemas![schemaName] = schema;
+    }
+  };
+
+  private queryParameters = () => {
+    const parameters = this.versionFile.components?.parameters;
+    if (typeof parameters === 'undefined') {
+      return {};
+    }
+
+    const schemas: Record<string, any> = {};
+    for (const [name, parameter] of Object.entries(parameters)) {
+      if (isReferenceObject(parameter)) {
+        continue;
+      }
+
+      if (parameter.in != 'query') {
+        continue;
+      }
+
+      const schema = {
+        type: 'object',
+        properties: Object.assign({}, { [parameter.name]: parameter.schema }),
+      };
+      schemas[name] = schema;
+    }
+
+    return schemas;
+  };
+
+  private copyQueryParametersToSchemas = (): void => {
+    const parameters = this.queryParameters();
+    for (const [name, schema] of Object.entries(parameters)) {
+      this.versionFile.components!.schemas![name + 'QueryParameter'] = schema;
+    }
+  };
+
+  public generateTypesFromSchemas = async (
+    schemaFilter?: StringFilter
+  ): Promise<string[]> => {
+    if (typeof this.versionFile.components?.schemas === 'undefined') {
+      throw new Error('components.schema is undefined');
+    }
+
+    if (typeof schemaFilter !== 'undefined') {
+      const schemas = this.versionFile.components!.schemas!;
+      this.versionFile.components!.schemas = Object.keys(schemas)
+        .filter(schemaFilter)
+        .reduce(
+          (acc, schemaName) =>
+            Object.assign(acc, { [schemaName]: schemas[schemaName] }),
+          {}
+        );
+    }
+
+    const typeNames = Object.keys(this.versionFile.components?.schemas || {});
+    await this.runSwaggerGenerator();
+
+    return typeNames;
+  };
+
+  /**
+   * generateTypes
+   *
+   * generate all types with a given versionfile and configuration. Returns name of all generated types afterwards.
+   */
+  public generateTypes = async (): Promise<string[]> => {
+    this.filterPaths();
+    this.removeSchemasWithoutPathRelations();
+    this.copyResponsesToSchemas();
+    this.copyQueryParametersToSchemas();
+
+    for (const [typeName, schema] of this.autoNameInlinedResponses()) {
+      if (typeName in this.versionFile.components!.schemas!) {
+        console.error(
+          `unable to use automatic name for inlined response type. "${typeName}" already exists in the component schemas`
+        );
+        process.exit(1);
+      }
+
+      this.versionFile.components!.schemas![typeName] = schema;
+    }
+
+    for (const [typeName, schema] of this.autoNameInlinedRequests()) {
+      if (typeName in this.versionFile.components!.schemas!) {
+        console.error(
+          `unable to use automatic name for inlined request type. "${typeName}" already exists in the component schemas`
+        );
+        process.exit(1);
```

Comment:
I think throwing an exception is better here - since you're not really in the CLI level here